app/models/recipe.rb

```diff
@@ -1,3 +1,11 @@
 class Recipe < ApplicationRecord
-   validates :name, :complexity, :genre, presence: true
+   has_many :ingredients
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
```

Comment:
Remember that when adding a `has_many` relationship, it's because this table's primary key is a foreign key on the child table. I do not recall there being a foreign key for a recipe on the ingredients table. Furthermore, since this IC was built around a many-to-many relationship, the only way that a recipe could have many ingredients is through a joins table. So because you have like 4, connecting a recipe and ingredient through the recipe_ingredients table, you do not need line 2, that cannot be made since there is no direct connection between a recipe and ingredient.

The tests for the association were already created, all you needed to do was add those associations in their respective models to get all the tests to pass.